tests/test_internals/test_roi_extractors_module.py

```diff
@@ -230,6 +242,243 @@ def test_add_two_imaging_planes(self):
+
+        image_segmentation = ophys.data_interfaces.get(self.image_segmentation_name)
+        self.assertEqual(image_segmentation.name, self.image_segmentation_name)
+
+
+class TestAddPlaneSegmentation(unittest.TestCase):
+    def setUp(self) -> None:
+        self.num_rois = 10
+        self.num_frames = 20
+        self.num_rows = 25
+        self.num_columns = 20
+        self.segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=self.num_rois,
+            num_frames=self.num_frames,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+        self.session_start_time = datetime.now().astimezone()
+        self.nwbfile = NWBFile(
+            session_description="session_description",
+            identifier="file_id",
+            session_start_time=self.session_start_time,
+        )
+
+        self.metadata = dict(Ophys=dict())
+
+        self.image_segmentation_name = "image_segmentation_name"
+        self.plane_segmentation_name = "plane_segmentation_name"
+
+        self.plane_segmentation_metadata = dict(
+            name=self.plane_segmentation_name,
+            description="Segmented ROIs",
+        )
+
+        image_segmentation_metadata = dict(
+            ImageSegmentation=dict(
+                name=self.image_segmentation_name,
+                plane_segmentations=[
+                    self.plane_segmentation_metadata,
+                ],
+            )
+        )
+
+        self.metadata["Ophys"].update(image_segmentation_metadata)
+
+    def test_add_plane_segmentation(self):
+        """Test that add_plane_segmentation method adds a plane segmentation to the nwbfile
+        specified by the metadata."""
+        add_plane_segmentation(
+            segmentation_extractor=self.segmentation_extractor,
+            nwbfile=self.nwbfile,
+            metadata=self.metadata,
+        )
+
+        image_segmentation = self.nwbfile.processing["ophys"].get(self.image_segmentation_name)
+        plane_segmentations = image_segmentation.plane_segmentations
+
+        self.assertEqual(len(plane_segmentations), 1)
+
+        plane_segmentation = plane_segmentations[self.plane_segmentation_name]
+
+        self.assertEqual(plane_segmentation.name, self.plane_segmentation_name)
+        self.assertEqual(plane_segmentation.description, self.plane_segmentation_metadata["description"])
+
+        plane_segmentation_num_rois = len(plane_segmentation.id)
+        self.assertEqual(plane_segmentation_num_rois, self.num_rois)
+
+        plane_segmentation_roi_centroid_data = plane_segmentation["RoiCentroid"].data
+        expected_roi_centroid_data = np.array(self.segmentation_extractor.get_roi_locations()).T
+
+        assert_array_equal(plane_segmentation_roi_centroid_data, expected_roi_centroid_data)
+
+        image_mask_iterator = plane_segmentation["image_mask"].data
+
+        data_chunks = np.zeros((self.num_frames, self.num_columns, self.num_rows))
```

Comment:
I did not know myself how to extract data from the iterators orderly. Good to know.